ksqldb-parser/src/test/java/io/confluent/ksql/parser/AstBuilderTest.java

```diff
@@ -285,25 +288,43 @@ public void shouldIncludeSelectAliasIfPresent() {
   @Test
   public void shouldBuildLambdaFunction() {
     // Given:
-    final SingleStatementContext stmt = givenQuery("SELECT X => X + 5 FROM TEST1;");
+    final SingleStatementContext stmt = givenQuery("SELECT TRANSFORM_ARRAY(Col4, X => X + 5) FROM TEST1;");
 
     // When:
     final Query result = (Query) builder.buildStatement(stmt);
 
     // Then:
     assertThat(result.getSelect(), is(new Select(ImmutableList.of(
         new SingleColumn(
-            new LambdaFunctionExpression(
-                ImmutableList.of("X"),
-                new ArithmeticBinaryExpression(
-                    Operator.ADD,
-                    column("X"),
-                    new IntegerLiteral(5))
+            new FunctionCall(
+                FunctionName.of("TRANSFORM_ARRAY"),
+                ImmutableList.of(
+                    column("COL4"),
+                    new LambdaFunctionCall(
+                        ImmutableList.of("X"),
+                        new ArithmeticBinaryExpression(
+                            Operator.ADD,
+                            column("X"),
+                            new IntegerLiteral(5))
+                  )
+                )
             ),
             Optional.empty())
     ))));
   }
 
+  @Test
```

Comment:
Since in g4 we allow multiple expressions followed by multiple lambda functions, could we also add a test for the error case where number of expressions is not equal to the number of lambda functions?

Reply:
I don't think we can make this restriction, not all invocation functions would fit this requirement. From KLIP-30
```
reduce_map(map, s, (k, v, s) => s) - Reduces the input Map down to a single value. s is the initial state and is passed into the scope of the lambda function. Each invocation returns a new value for s, which the next invocation will receive. reduce_map will return the final value of s.
```
^ this one wouldn't fit this restriction

Reply:
Got it, thanks.